packages/labs/src/scoped-registry/scoped-registry-mixin.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * @license
+ * Copyright (c) 2018 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {LitElement} from 'lit-element';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type LitElementConstructor = new (...args: any[]) => LitElement;
+
+export type ScopedElementsMap = {
+  [key: string]: typeof HTMLElement;
+};
+
+export function UseScopedRegistry<SuperClass extends LitElementConstructor>(
+  superclass: SuperClass
+): SuperClass {
+  return class ScopedRegistryMixin extends superclass {
+    registry?: CustomElementRegistry;
+
+    /**
+     * Obtains the scoped elements definitions map
+     */
+    static get scopedElements(): ScopedElementsMap {
+      return {};
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    constructor(...args: any[]) {
+      super(...args);
+
+      this.registry = new CustomElementRegistry();
```

Comment:
I think all instances of an element definition should share a single registry by default, especially seeing as how the `scopedElements` property is static.

Reply:
And couldn't problems arise later with lazy definitions if we share the registry between different instances?

Reply:
oh, I miss "by default". We could always declare a registry per instance if needed. Makes sense.

Reply:
Done